kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/ServiceDiscovery.scala

```diff
@@ -257,6 +240,30 @@ object ServiceDiscovery extends Logging {
     }
     serviceNode
   }
+
+  def getEngineByPolicy(
```

Comment:
we don't need this, this has already been done at `Some("engine-pool-" + Random.nextInt(poolSize))`, we can use the old logic to get or create engine directly

Reply:
Make sense.